### PR TITLE
Plumb pylibcudf.io.parquet options classes through cudf python

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Apart of #15162

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/_lib/parquet.pyx [hunk 1]
```diff
@@ -442,44 +430,36 @@ def write_parquet(
...
-    cdef sink_info sink = make_sinks_info(
-        filepaths_or_buffers, _data_sinks
-    )
-
     if index is True or (
         index is None and not isinstance(table._index, cudf.RangeIndex)
     ):
-        tv = table_view_from_table(table)
-        tbl_meta = table_input_metadata(tv)
+        columns = table._columns if table._index is None else [
```
Comment (mroeschke) [inline python/cudf/cudf/_lib/parquet.pyx:436]: `table._index` should (ideally) never be `None`. Is there a test failure if we don't check for this?

File: python/cudf/cudf/_lib/parquet.pyx [hunk 2]
```diff
@@ -675,52 +631,31 @@ cdef class ParquetWriter:
                 table,
                 num_partitions=len(partitions_info) if partitions_info else 1
             )
-
-        cdef table_view tv
         if self.index is not False and (
             table._index.name is not None or
                 isinstance(table._index, cudf.core.multiindex.MultiIndex)):
-            tv = table_view_from_table(table)
+            columns = table._columns if table._index is None else [
```
Comment (mroeschke) [inline python/cudf/cudf/_lib/parquet.pyx:637]: Same question here

File: python/pylibcudf/pylibcudf/io/parquet.pyx [hunk 3]
```diff
@@ -246,6 +250,288 @@ cpdef read_parquet(
...
+
+        Returns
+        -------
+        None
+        """
+        cdef vector[string] column_chunks_file_paths
+        cdef unique_ptr[vector[uint8_t]] out_metadata_c
+        if metadata_file_path:
+            for path in metadata_file_path:
+                column_chunks_file_paths.push_back(str.encode(path))
```
Comment (mroeschke) [inline python/pylibcudf/pylibcudf/io/parquet.pyx:271]:
```suggestion
                column_chunks_file_paths.push_back(path.encode())
```

Nit

File: python/cudf/cudf/_lib/parquet.pyx (hunk 1 above)
Comment (Matt711) [reply python/cudf/cudf/_lib/parquet.pyx]: Thanks, no test failures. I removed the checks.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**